activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -31,6 +31,9 @@ class CollectionProxy < Relation
       def initialize(klass, association) #:nodoc:
         @association = association
         super klass, klass.arel_table, klass.predicate_builder
+
+        extensions = association.extensions
+        extend(*extensions) if extensions.any?
```

Comment:
Looks like we can shave a little bit more: https://gist.github.com/matthewd/d942b8dd3e55f17e2e1850d73b5c3e76

I agree we can then skip caching; it seems close enough, and the cut-down `extensions` method reminds us that the answer *could* vary between owners, given a sufficiently creative scope proc.